### ISSUE TITLE
Any solution to customize `node-linker` setting?

### ISSUE THREAD

**Issue by sneko** [TheSecurityDev/heroku-buildpack-nodejs-pnpm#2]
Hi @TheSecurityDev ,

I'm trying to make my final image the smallest possible. Since I'm using `next.js` I build my project as `standalone` so it embeds only the needed dependencies from `node_modules` into the build (but it's still a structure of modules, there are not minified or so...). Like that, I should be able to remove my root `node_modules` (and also those for each monorepo package) that takes a huge amount of space in the final image.

Unfortunately, since by default `pnpm` uses symbolic links, removing the root `node_modules` breaks things (my standalone build having copied simlinks targeting it). I would like to have your buildpack installs with `node-linker=hoisted` as specified into https://pnpm.io/npmrc#node-linker .

Since there is no way to customize this an environment variable, I could fork your repo to force using `hoisted` but do you see something possible with your actual code?

Any suggestion appreciated :) 

Thank you,

(from what I understand `node-linker=hoisted` won't solve this because this setting is about the project root `node_modules`)

**Comment by TheSecurityDev** [comment]
I'm not completely sure, but I think you can create a .npmrc config file with the hoist setting you need.

https://pnpm.io/npmrc

Edit: I see that's the same link you provided, so maybe I'm misunderstanding something.

**Comment by sneko** [comment]
Creating a `.npmrc` file was not ideal because at the time I wrote the issue I wanted to have `hoisted` in my CI/CD, but still the default behavior locally (that's why I searched for environment variable or so).

At the end I tested multiple buildpack settings by forking yours, and just end up with a solution without using `hoisted` 👍 

I had an issue with workspace settings as explained in https://github.com/orgs/pnpm/discussions/6337#discussioncomment-5499419

Also, I changed how I optimize my final image, I build a standalone Next.js version of my project (it will embed only needed `node_modules` in the `dist` folder), then I can remove all other `node_modules`. Like that no need to deal with prune dev dependencies that can also complicates the split logic of prod/dev deps (it helped also save a few minutes I think).

For record:
* https://github.com/inclusion-numerique/mediature/commit/d8c0f421679f0c14c7f1ab59aaebcd70602c5c80
* https://github.com/sneko/heroku-buildpack-nodejs-pnpm/tree/pruning-disabled

Thank you for your time,
